sklearn/datasets/_svmlight_format_io.py

```diff
@@ -393,39 +380,34 @@ def _dump_svmlight(X, y, f, multilabel, one_based, comment, query_id):
 
         f.write(b"#\n")
         f.writelines(b"# %s\n" % line for line in comment.splitlines())
-
-    for i in range(X.shape[0]):
-        if X_is_sp:
-            span = slice(X.indptr[i], X.indptr[i + 1])
-            row = zip(X.indices[span], X.data[span])
-        else:
-            nz = X[i] != 0
-            row = zip(np.where(nz)[0], X[i, nz])
```

Comment:
What does line profiling say about `main`'s implementation? The the dense X case, this looks like the place this PR is optimizing for. In the other words, where is the performance coming from with the dense/dense case?

BTW, on `main`, a more optimal approach is:

```python
nz_idx = np.flatnonzero(X)
s = " ".join(value_pattern % (n, X[n]) for n in nz_idx)
```